### ISSUE TITLE
Enable HybridApp test builds in App repo

### ISSUE THREAD

**Issue by Julesssss** [Expensify/App#51636]
### Problem

During PR review we perform test only against ND while the users will use Hybrid App. It prevent us from finding possible bugs and problem users will face.

_P/S from Jan [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1730113645718229)_

### Solution

Let's configure our CI/CD so it's possible to trigger AdHoc build of Hybrid App from ND repo.

**Comment by jnowakow** [comment]
Hello, I'm from SoftwareMansion and I'll work on this

**Comment by jnowakow** [comment]
@Julesssss I managed to successfully add new build type for Android and new scheme for iOS - [here's draft PR](https://github.com/Expensify/Mobile-Expensify/pull/13258). 
The next step requires action from someone who has access to Firebase console and developer profile because I think it's time to generate new `google-services.json`, `GoogleService-info.plist` and create new provisioning profile. 

I think that bundle id's are required for that configuration so I propose:
iOS: 
`com.expensify.expensifylite` - Release
`com.expensify.expensifylite.debug` - Debug
`com.expensify.expensifylite.adhoc` - AdHoc

Android: 
`org.me.mobiexpensifyg` - release
`org.me.mobiexpensifyg.debug` - debug
`org.me.mobiexpensifyg.adhoc` - adhoc

Please let me know if I should provide some additional info and when this can be accomplished

**Comment by Julesssss** [comment]
Hey, sorry for the delay! I will make time for this as soon as possible. No extra details required yet.

**Comment by jnowakow** [comment]
No problem, I've started to work on other things in the meantime :)

**Comment by Julesssss** [comment]
Sorry for the delay here, I'll have time to review tomorrow!

**Comment by Julesssss** [comment]
Hey @jacobkim9881, I can generate the Firebase apps. I'm thinking we should do `.dev` instead of `.debug` to align with `App`, and to make it clear that these are dev builds. 

Let me know if you see any problems with this

**Comment by Julesssss** [comment]
Hey @jnowakow, I can generate the Firebase apps. I'm thinking we should do `.dev` instead of `.debug` to align with `App`, and to make it clear that these are dev builds. 

Let me know if you see any problems with this

**Comment by Julesssss** [comment]
I have configured the iOS/Android apps in Firebase (using `.dev`). I haven't yet added the Android SHA certificate fingerprints, so Google SSO and other things won't work. But I think that is fine. 

@jnowakow I will DM you the google services file.

**Comment by jnowakow** [comment]
@Julesssss I `.dev` instead of `.debug` is great for me. Thanks for taking care of it!

**Comment by Julesssss** [comment]
Hey @jnowakow, we've been chatting 1:1 but could you share an update on this issue please?

**Comment by jnowakow** [comment]
Sure!
Last update is that everything seems to work fine with provided firebase configurations. In order to proceed with iOS I'll need new provisioning profile, notification service profile and certificates. For Android I think I don't need any additional set up so I'll focus right now on creating GH workflow

**Comment by jnowakow** [comment]
@Julesssss here are the current changes:
[old dot](https://github.com/Expensify/Mobile-Expensify/pull/13297)
[new dot](https://github.com/Expensify/App/pull/53214)
I think everything is in place to test if android adhoc build is working fine

**Comment by Julesssss** [comment]
Nice, I will take a look. 

I assume we need to merge Mobile-Expensify first? If so can we put a hold on the App PR? And mark Mobile-Expensify as ready for review. I can also try to run a test build against this branch if we are ready.

**Comment by jnowakow** [comment]
Thanks for review! I've answered the comments and addressed those two little TODOs. 
In general yes, I think we would have to merge OD first but iOS is not 100% ready (I think nothing should break). I'm also afraid that we have to merge this PR in order to be able to run this workflow

**Comment by jnowakow** [comment]
btw I think there is a possible improvement to this workflow to enable people to specify old dot's commit they want to build app from. It would enable building hybrid app not only with ND changes that are not merged but also with OD changes which weren't yet merged to main

**Comment by Julesssss** [comment]
Okay, I will try running the action when I have a moment. 

The improvement could be good, let's discuss that after the initial PRs.

**Comment by Julesssss** [comment]
Mobile PR merged.

**Comment by Julesssss** [comment]
Attempting to test the first of the workflows [here](https://github.com/Expensify/App/pull/53214#issuecomment-2511174489).

**Comment by Julesssss** [comment]
Initial 2 [PRs](https://github.com/Expensify/App/pull/53214) merged today. We'll need to await deployment to prod for testing...

**Comment by Julesssss** [comment]
^ waiting for [prod deployment](https://github.com/Expensify/App/pull/53214#issuecomment-2513031223) in order to test above PRs.

**Comment by Julesssss** [comment]
Build seemed to be [successful](https://github.com/Expensify/App/actions/runs/12137896092/job/33844021159#step:5:7), we has a post build permission error with build artifacts.

**Comment by jnowakow** [comment]
Yeah, app build correctly but there was problem with uploading. I'll take a look why this path is wrong

**Comment by Julesssss** [comment]
Second PR was merged and we're running another attempt [here](https://github.com/Expensify/App/pull/53530).

**Comment by jnowakow** [comment]
@Julesssss do you have some timeline when I can get provisioning profile, notification service profile and certificates in order to proceed with iOS?

**Comment by Julesssss** [comment]
Hey, I can get the profiles but can you just remind me whether **you** will need to register your phone and profile? We are currently limited and being ignore by Apple.

**Comment by jnowakow** [comment]
I hope registering my phone and profile will not be necessary

**Comment by Julesssss** [comment]
Great thanks. I sent you the provisioning profiles and I need to think about how we provided other certificates.

**Comment by Julesssss** [comment]
I added/updated the AdHoc iOS profiles in 1Password. Shared names with Jan.

**Comment by Julesssss** [comment]
Initial iOS PR was merged, once deployed we can kick off a test build.

**Comment by Julesssss** [comment]
Another PR [here](https://github.com/Expensify/App/pull/54034) to apply changes necessary due to our submodule changes.

**Comment by Julesssss** [comment]
- [Error](https://github.com/Expensify/App/actions/runs/12300037696/job/34327243384#step:13:6095) with Android HybridApp build
    - Sounds similar to a fix we made [here](https://github.com/Expensify/Mobile-Expensify/pull/13329/files)
- [Error](https://github.com/Expensify/App/actions/runs/12300037696/job/34327243800#step:13:13) with iOS HybridApp build
    - OldApp_AdHoc with 1Pass filename

**Comment by jnowakow** [comment]
[Here](https://github.com/Expensify/App/pull/53954#issuecomment-2539280012) are results from workflow run after updating it with new repos structure. Android failed because the branch is not in sync with main. It's because `Fastfile` on the brach doesn't contain necessary changes. Android should work fine on branches that are in sync with newest main. 
On iOS there was problem with getting documents from 1Password. @Julesssss could you check if you are able to run this commands locally?
```
op document get --output ./OldApp_AdHoc.mobileprovision OldApp_AdHoc
op document get --output ./OldApp_AdHoc_Share_Extension.mobileprovision OldApp_AdHoc_Share_Extension
op document get --output ./OldApp_AdHoc_Notification_Service.mobileprovision OldApp_AdHoc_Notification_Service
```

**Comment by Julesssss** [comment]
Hi @jnowakow, yep all three of these work correctly for me. Modifying the names causes errors.

**Comment by jnowakow** [comment]
Hmm then I don't know where the problem is 🤔 I copied those commands from workflow so I don't know where I could do the typo

**Comment by jnowakow** [comment]
@Julesssss 
I see plan for proceeding with this issue as follows:

Merge PR with changes for iOS in [OD](https://github.com/Expensify/Mobile-Expensify/pull/13297). I've merged newest main and adjusted it to new repo structure. I've tested it locally and all three schemas seems to work fine. 

Figure out why 1Password [doesn't see apple related files](https://github.com/Expensify/App/actions/runs/12300037696/job/34327243800#step:13:14) that you added recently. I'm afraid I need you help here because I don't have access to 1Pass. Can you check if those files are added in the same way as other files (eg. `upload-key.keystore` for Android)? Maybe they have to be exported or something like that? 

Figure out why `validateActor` [is failing](https://github.com/Expensify/App/actions/runs/12354702278/job/34476645492). I suspect it was some random issue and if we rerun the workflow this step should pass. This step wasn't modified in the meantime and was working fine before. It's also the same as in other workflows (eg. [here](https://github.com/Expensify/App/blob/f6bba1f5f9a4e034b585715a2da8c3104274b797/.github/workflows/testBuild.yml#L17-L41)) and there it isn't failing. If we rerun the workflow and is keeps failing then I have to investigate it. 

I think that on ND side almost everything is ready. There are small corrections to do like fixing the typo in workflow name or running dev scheme when launching from terminal.

**Comment by Julesssss** [comment]
> Merge PR with changes for iOS in https://github.com/Expensify/Mobile-Expensify/pull/13297

Reviewing now 👍 

> Can you check if those files are added in the same way as other files (eg. upload-key.keystore for Android)? Maybe they have to be exported or something like that?

Sure, I am rushing to finish some things before going OOO but I can help with that in the new year 

> Figure out why validateActor [is failing](https://github.com/Expensify/App/actions/runs/12354702278/job/34476645492). I suspect it was some random issue and if we rerun the workflow this step should pass

Also sounds good

**Comment by jnowakow** [comment]
@Julesssss I've opened [PR](https://github.com/Expensify/Mobile-Expensify/pull/13350/files) which restore changes reverted before Christmas because of problems with publishing iOS to AppStore/TestFlight. I think the problem was caused by embedding `Pods_NotificationServiceExtension.framework`. Are you able to build app locally and submit it to TestFlight? It would be great to test it to avoid another revert. 

Another issue is that I've checked provisioning profiles you generated. I think the bundle ids there are wrong :/ Bundle Ids right now are `com.expensify.expensifylite.adhoc`, `com.expensify.expensifylite.adhoc.SmartScanExtension` (provisioning profile is called Share Extension) and `com.expensify.expensifylite.adhoc.NotificationServiceExtension`.

**Comment by Julesssss** [comment]
Thanks @jnowakow, I see Vit merged the PR. 

As for the other things I will check hopefully later this week. Or perhaps next week.

**Comment by Julesssss** [comment]
> Are you able to build app locally and submit it to TestFlight?

As the [PR](https://github.com/Expensify/Mobile-Expensify/pull/13350) was merged, we just need to look out for the next few builds, I think.

> I think the bundle ids there are wrong 

No problem, so right now in the console I see that the bundle IDs are identical to the production bundleID. So I think I'll need to create new profiles to match the `.adhoc` IDs we have chosen, right?

**Comment by Julesssss** [comment]
Great progress has been made this week by @mountiny and @jnowakow 🎉 

I'm going to share the updates here, as the previous issue was specifically about the profiles. [Here](https://expensify.slack.com/archives/C04878MDF34/p1737729414280249?thread_ts=1737502593.987369&cid=C04878MDF34) is a long thread with progress updates. We're now at the point where the workflow has permission to view the profiles, and we're updating them to include all necessary capabilities.

[Here's](https://github.com/Expensify/App/actions/runs/12955519236) the latest run, which is built using changes from [this](https://github.com/Expensify/App/pull/55615) App PR, and [this](https://github.com/Expensify/Mobile-Expensify/pull/13370) Mobile-Expensify PR.

**Comment by Julesssss** [comment]
Capabilitiy issues were resolved and profiles updated. Next [error](https://github.com/Expensify/App/actions/runs/12955966695/job/36141246180#step:16:23723) is about the profile type not being for iOS App Store.

NewDot standalone AdHoc profiles also used ad-hoc though, so this should be solvable without regeneration 🤞

**Comment by Julesssss** [comment]
To solve the above problem we just needed to set the Fastlane type to ad-hoc: 

<img width="552" alt="Image" src="https://github.com/user-attachments/assets/8996c947-f205-4dbc-9d07-cde8b39c9c72" />

I'm not able to push to @jnowakow's branch, so I cloned the branch [here](https://github.com/Expensify/App/pull/55751). [Here's](https://github.com/Expensify/App/actions/runs/12956636582) the test run.

**Comment by Julesssss** [comment]
The above change resulted in a failure with the export_method not being picked up.  HOWEVER, building from @mountiny's branch with our additional changes led to a successful [build](https://github.com/Expensify/App/actions/runs/12957511114/job/36146048401). Successful QR code [comment](https://github.com/Expensify/App/pull/55577#issuecomment-2613426621).

<img width="833" alt="Image" src="https://github.com/user-attachments/assets/c70be73c-bd06-4017-be1c-6bc4683620bf" />

![Image](https://github.com/user-attachments/assets/49bf03f6-405d-4cf2-b19e-0abb877d74b1)

The app runs smoothly. Next step is to plan how to merge these PRs without breaking iOS dev builds.

**Comment by Julesssss** [comment]
- [Mobile-Expensify](https://github.com/Expensify/Mobile-Expensify/pull/13370) PR - can we strip out changes unrelated to AdHoc environment?
- [App PR](https://github.com/Expensify/App/pull/55577) - these seem pretty safe, but lets merge after Mobile to avoid any failing runs in the meantime

**Comment by Julesssss** [comment]
> * [Mobile-Expensify](https://github.com/Expensify/Mobile-Expensify/pull/13370/files) PR - can we strip out changes unrelated to AdHoc environment?
> * [App PR](https://github.com/Expensify/App/pull/55577) - these seem pretty safe, but lets merge after Mobile to avoid any failing runs in the meantime

@jnowakow @mountiny -- I think we can go ahead and merge the PRs. We've stripped out dev changes from the App PR [here](https://github.com/staszekscp/Mobile-Expensify/pull/15). I merged main into the App PR and ran a final test run [here](https://github.com/Expensify/App/actions/runs/12997105420), so we just need a review from people who aren't us three!

**Comment by mountiny** [comment]
Lets do it!

**Comment by Julesssss** [comment]
Hey @AndrewGable @arosiclair. Three of us committed changes related to iOS AdHoc builds, so a review of the App and Mobile-Expensify branches would be much appreciated. Other than the Mobile-Expensify iOS build changes they should be pretty safe as we're fixing a workflow that hasn't fully worked yet.

- https://github.com/Expensify/App/pull/55577
- https://github.com/Expensify/Mobile-Expensify/pull/13370

**Comment by Julesssss** [comment]
Both PRs are ready to go, ideally close together. I keep having to merge main as the submodule diff is incorrectly flagged as a conflict.

**Comment by jnowakow** [comment]
@Julesssss probably [this one](https://github.com/Expensify/App/actions/runs/12997157241) was final test run 😅

**Comment by Julesssss** [comment]
PRs merged. [CP was requested](https://expensify.slack.com/archives/C07J32337/p1738092439538809), but failed due to a 'conflict' with the submodule commitID. Though identical, the CP failed as it can't be auto-resolved I guess. We'll have to await regular deployment.

**Comment by jnowakow** [comment]
@Julesssss it seems that the solution is quite stable 🥳 

We can start thinking about follow-ups to make the flow more convenient to use. Here are some improvements that can be made:
- merge `Build and deploy apps for testing` and `Build and deploy hybrid apps for testing` 
- Pasting link to mobile-expensify PR instead of PR number
- Option to trigger build without open PR in ND. It would build from ND’s main and would post a comment with links in OD PR

**Comment by jnowakow** [comment]
I'm wondering about two things about adhoc builds. Why they should be run when PR to branch with `ci-test` in name are opened? And why `Ready To Build` label is necessary to run those workflows? It seems like it was created this way to make it easier for people from outside Expensify to test this workflow? 🤔 
Does somebody know the answer? @Julesssss @mountiny @AndrewGable @staszekscp

**Comment by Julesssss** [comment]
> Why they should be run when PR to branch with ci-test in name are opened?

Not sure about this one. 

> And why Ready To Build label is necessary to run those workflows? 

I believe this is prevent a malicious actor from being able to generate builds with any modification and without internal review. 


The suggested improvements sound great. Being able to run without an App PR would be my preferred priority. Also I'm not sure how much benefit we get from 'Pasting link to mobile-expensify PR instead of PR number'?

**Comment by AndrewGable** [comment]
cc @roryabraham

**Comment by staszekscp** [comment]
Yeah, the `ci-test` I think is a remaining from ancient history, when I was building the original solution - as far as I remember it was used by me to trigger builds automatically, without assistance of an Expensify employee. I think it stayed there because we had to do some follow-up tasks, and we've forgotten to remove it

**Comment by mountiny** [comment]
It would be nice if we did not have to add the App PR necessarily and just use App main with Mobile-Expensify PR number

**Comment by jnowakow** [comment]
Okey, then I think we can get rid of `ci-test` branches as they are not used anymore. Without them `Ready to build` label doesn't make sense anymore because only Expensify employees can trigger workflows. It will make workflow cleaner and easier to add option to run workflow without PR in App

**Comment by Julesssss** [comment]
> It would be nice if we did not have to add the App PR necessarily and just use App main with Mobile-Expensify PR number

Agree. @jnowakow shared a few follow-up improvements elsewhere (I can't find the issue currently). 

I'm going to close out this issue 🎉

**Comment by Julesssss** [comment]
> Okey, then I think we can get rid of ci-test branches as they are not used anymore. Without them Ready to build label doesn't make sense anymore because only Expensify employees can trigger workflows.

@jnowakow lets add this to the other issue for further discussion. I'll link it when I find it.

**Comment by Julesssss** [comment]
Further improvements being tracked here: https://github.com/Expensify/App/issues/56848
